contrib/checkstyle/src/mill/contrib/checkstyle/CheckstyleModule.scala

```diff
@@ -0,0 +1,183 @@
+package mill
+package contrib.checkstyle
+
+import mill.api.{Loose, PathRef}
+import mill.scalalib.{DepSyntax, JavaModule, ScalaModule}
+import mill.util.Jvm
+
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.stream.{StreamResult, StreamSource}
+
+trait CheckstyleModule extends JavaModule {
+
+  import CheckstyleModule._
+
+  def checkstyle: T[Seq[PathRef]] = T {
+
+    val dst = T.dest
+
+    val classpath = checkstyleClasspath().map(_.path)
+    val options = checkstyleOptions()
+    val config = checkstyleConfig()
+    val format = checkstyleFormat()
+    val report = dst / checkstyleReport()
+    val inputs = sources().map(_.path.toString())
+    val _throw = checkstyleThrow()
+    val transforms = checkstyleTransforms()
+    val transformer = checkstyleTransformer(format)
```

Comment:
These should be inlined and do not need to be lifted to the top of the block like this